internal/cmdrender/executor.go

```diff
@@ -316,8 +316,33 @@ func fnChain(pl *kptfilev1alpha2.Pipeline, pkgPath types.UniquePath) ([]kio.Filt
 	return runners, nil
 }
 
+// errorIfDuplicateAnnotation returns an error if the same index/path is on multiple resources
+func errorIfDuplicateAnnotation(nodes []*yaml.RNode) error {
+	// map has structure path -> index -> bool
+	// to keep track of paths and indexes found
+	pathIndexes := make(map[string]map[string]bool)
+	for _, node := range nodes {
+		filepath, index, err := kioutil.GetFileAnnotations(node)
+		if err != nil {
+			return err
+		}
+		filepath = strings.TrimPrefix(filepath, "./")
+		if pathIndexes[filepath] == nil {
+			pathIndexes[filepath] = make(map[string]bool)
+		}
+		if _, ok := pathIndexes[filepath][index]; ok {
+			return fmt.Errorf("duplicate path and index %s %s", filepath, index)
+		}
+		pathIndexes[filepath][index] = true
+	}
+	return nil
+}
+
 // trackInputFiles records file paths of input resources in the hydration context.
 func trackInputFiles(hctx *hydrationContext, input []*yaml.RNode) error {
+	if err := errorIfDuplicateAnnotation(input); err != nil {
```

Comment:
I would like this logic to be executed after each function is run in the pipeline to catch the error early and also determine source of the problem to help user. That will require wrapping the `FunctionFilter`. But, this is a good starting point for now, and this logic is portable because it acts on `[]*yaml.RNode`, so we can do that later.

Reply:
That makes sense. Originally I was trying to do it in kyaml but then realized we would have to wait for release. Let me know when we'd like to move it. I can work on that.